run-checks

```diff
@@ -394,27 +394,34 @@ if [ "$STATIC" = 1 ]; then
       exit 1
     fi
 
-    if [ -z "${SKIP_GOLANG_CI_LINT-}" ]; then
+    if [ -z "${SKIP_GOLANGCI_LINT-}" ]; then
 
         echo "Checking installation of golangci-lint"
-        if ! command -v golangci-lint > /dev/null ; then
-            echo "ERROR: Cannot find golangci-lint. You need to first install the golangci-lint snap"
+        gcil="$(command -v golangci-lint || true)"
+        if [ -z "$gcil" ]; then 
+            echo "ERROR: Cannot find golangci-lint. You need to first install the golangci-lint"
             exit 1
         fi
-        if snap refresh --list | grep -q golangci-lint; then
-            echo "WARNING: your golangci-lint snap is out of date. The CI will install a fresh version, which may differ from yours."
-        fi
-        if ! snap list golangci-lint | grep -q latest; then
-            echo "WARNING: your golangci-lint snap is not installed from the latest/* channel."
+
+        if [ "${gcil/\/snap\/bin/}" != "$gcil" ]; then
```

Comment:
Nitpick: I'm not familiar with this construction, is this a bash-ism for running a replacement on a variable? If this is bash-specific, I'd stick to POSIX shell and do something a bit more explicit. If this is supported by POSIX-compliant shells then (oh god) it's fine, but I'd still personally prefer something easier to read.

Edit: looks like zsh supports this, but dash doesn't seem to like it:
```
 oac@xps15  ~  foo="hello"            
 oac@xps15  ~  echo $foo        
hello
 oac@xps15  ~  echo "${foo/ll/}"
heo
 oac@xps15  ~  echo "${foo/ll/aa/}"
heaa/o
 oac@xps15  ~  echo "${foo/ll/aa}" 
heaao
 oac@xps15  ~  dash                                                      
$ foo="hello"
$ echo $foo
hello
$ echo "${foo/ll/aa}"
dash: 3: Bad substitution
$ echo "${foo/ll/}"
dash: 4: Bad substitution
```

Not sure what the takeaway is from this... I think I'd suggest to just do this instead:
```suggestion
        if echo "$gcil" | grep '^\/snap\/bin\/' ; then
```

Reply:
it's likely that `${var/<pattern>/<subst>}` works everywhere but `dash`, alternative would be to use:

```
"${gcil%/snap/bin/golangci-lint}"
```

which should work in dash too, or:
```
if echo "$gcil" | grep '/snap/bin/' ; then
```

(without `^` so that it matches both /snap/bin and /var/lib/snapd/snap/bin)